### PR TITLE
Add List<> support to the kotlin schema code generator

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: java/arcs/core/data/util/ReferencableList.kt [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import kotlin.reflect.KClass
+
+/**
+ * Represents a list of primitives which can be referenced - and thus used
+ * as field values by CRDT Collections & Singletons.
+ */
+data class ReferencableList<T>(
+    /** Type of primitive inside the list being converted into a referencable */
+    private val klass: KClass<*>,
+    /** The value */
```
Comment (csilvestrini) [inline java/arcs/core/data/util/ReferencableList.kt:25]: You can just delete this comment, it isn't helpful.

File: java/arcs/sdk/wasm/Encoding.kt [hunk 2]
```diff
@@ -79,6 +79,9 @@ class StringDecoder(private var bytes: ByteArray) {
     fun decodeDouble(): Double =
         throw NotImplementedError("No JS to Kotlin encoding exists for Kotlin Doubles")
 
+    fun <T> decodeList(): List<T> =
+        throw NotImplementedError("No JS to Kotlin enocding exists for Kotlin Lists")
```
Comment (shaper) [inline java/arcs/sdk/wasm/Encoding.kt:83]: nit: typo in 'enocding'

File: java/arcs/core/data/util/ReferencablePrimitive.kt [hunk 3]
```diff
@@ -56,8 +56,8 @@ data class ReferencablePrimitive<T>(
         private const val primitiveKotlinString = "kotlin.String"
         private const val primitiveKotlinBoolean = "kotlin.Boolean"
         private const val primitiveKotlinByteArray = "kotlin.ByteArray"
-        private val primitiveKClassMap = mapOf<KClass<*>, String>(
-            Byte::class to primitiveKotlinChar,
+        public val primitiveKClassMap = mapOf<KClass<*>, String>(
+            Byte::class to primitiveKotlinByte,
```
Comment (shaper) [inline java/arcs/core/data/util/ReferencablePrimitive.kt:60]: nit: IMO worth noting this fix in commit description as well.

Comment (shans) [reply java/arcs/core/data/util/ReferencablePrimitive.kt:60]: will do

**Pull request merged**